src/inject/dynamic-theme/watch.ts

```diff
@@ -37,24 +37,24 @@ function collectUndefinedElements(root: ParentNode) {
         });
 }
 
-function customElementsWhenDefined(tag: string) {
+const resolvers = new Map<string, () => void>();
+
+const handleIsDefined = (e: any) => {
```

Comment:
Worth using type `CustomEvent<{tag: string}>`.

I also suggest you to think about replacing `const` with `function`. Some would say that `const` is not overridable, uses a proper `this` scope when defined in classes, but I find it more readable and shorter, especially in root scope. `() => {}` are good when they are one-liners or passed as an argument into another function.

Funny article about this topic https://www.freecodecamp.org/news/constant-confusion-why-i-still-use-javascript-function-statements-984ece0b72fd/

Reply:
>  but I find it more readable and shorter, especially in root scope. `() => {}` are good when they are one-liners or passed as an argument into another function.

I agree with you on this and that's why I used a `const` for this one as it's going to be passed.